pkg/download/blobwithmsitoken.go

```diff
@@ -0,0 +1,116 @@
+package download
+
+import (
+	"fmt"
+	"github.com/Azure/azure-extension-foundation/httputil"
+	"github.com/Azure/azure-extension-foundation/msi"
+	"github.com/pkg/errors"
+	"net/http"
+	url2 "net/url"
+	"strings"
+)
+
+const (
+	xMsVersionHeaderName = "x-ms-version"
+	xMsVersionValue      = "2018-03-28"
+	storageResourceName  = "https://storage.azure.com/"
+)
+
+var azureBlobDomains = map[string]interface{}{ // golang doesn't have builtin hash sets, so this is a workaround for that
+	"blob.core.windows.net":       nil,
```

Comment:
shouldn't it be true instead of nil ?

Reply:
Its inconsequential, I am only using the presence/absence of the key in the map/dictionary. I am not using the value.